### ISSUE TITLE
Does YOLO v8 classification not require dataset? I don't get it how to do train yolov8 classifier by custom dataset. 

### ISSUE THREAD

**Issue by udkii** [ultralytics/ultralytics#16729]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and [discussions](https://github.com/ultralytics/ultralytics/discussions) and found no similar questions.

### Question

Hello. I am trying to train yolov8 classifier but I don't get it how to do. I only used YOLO for object detection, but not for classification. So I'm confused about preparing custom dataset. 

I saw solved question #1101 , and the answer was 'there's no need for dataset'. 
But if classification task doesn't need the dataset file, then how to specify number of classes and their names? 

When I did object detection, I labeled every objects for train set in coco dataset format. 
And now, I want to classify the document types by using YOLO. 

So here's my questions. 
1. Do I have to label all of my documents whether they are email or not? If so, how can I do? If not, then how can YOLO know the correct classes?
2. How can I make yaml file? There's no default YAML file in the ultralytics folder. 
3. How can I create train / val / test dataset for YOLO v8 classifier?

I need someone's help. 
Thank you so much.

**Comment by UltralyticsAssistant** [comment]
👋 Hello @udkii, thank you for reaching out to Ultralytics 🚀! This is an automated response to guide you through some common questions, and an Ultralytics engineer will assist you soon.

For a better understanding of YOLOv8 classification with custom datasets, we recommend checking our [Docs](https://docs.ultralytics.com) where you'll find relevant [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) examples.

If this involves a specific 🐛 Bug, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) so we can assist more effectively.

### Answering Your Questions
1. **Labeling:** For classification, your documents need to be labeled according to their class (e.g., email, not email).
2. **YAML File:** You can define your dataset structure in a YAML file. Here's a helpful guide on [dataset configuration](https://docs.ultralytics.com/yolov5/tutorials/train_custom_data/).
3. **Train/Val/Test Dataset:** You need to split your data into these sets. Each class should have its own folder with images.

Join our Ultralytics community for further discussions:
- [Discord](https://ultralytics.com/discord) 🎧 for real-time chat
- [Discourse](https://community.ultralytics.com) for in-depth discussions
- [Subreddit](https://reddit.com/r/ultralytics) for community interaction 

## Upgrade

Make sure your setup is up-to-date with the latest release. Upgrade using:

```bash
pip install -U ultralytics
```

## Environments

Consider running YOLO in verified environments with dependencies pre-installed:

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/ultralytics/yolov8"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image:** [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>

Ensure all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. This checks for correct operation on macOS, Windows, and Ubuntu every 24 hours and on every commit.

We hope these resources help you on your YOLOv8 journey! 😊

**Comment by udkii** [comment]
> Classification doesn't use yaml file. The class names are obtained from the folder name that the image files are in.
> 
> See the structure in the docs: https://docs.ultralytics.com/datasets/classify/

Thank you! Now I got it.

**Comment by glenn-jocher** [comment]
You're welcome! If you have any more questions, feel free to ask.

**Comment by UltralyticsAssistant** [comment]
A potential fix for this issue has been merged in [PR #22192: Verify passed dataset is a directory and not a file for classification training](https://github.com/ultralytics/ultralytics/pull/22192) by @Y-T-G with contributions from @glenn-jocher.

What changed and why this may resolve the issue:
- The classification dataset checker now validates that the provided path is a directory.
- If a file-like path is given (e.g., has a suffix), a clear ValueError is raised with guidance instead of attempting an auto-download.
- This avoids confusing behavior and helps ensure classification training uses the expected directory structure.
- See the [Classification dataset docs](https://docs.ultralytics.com/datasets/classify/) for the required layout.

How to test the fix:
- Install from the latest main branch:
  - `pip install -U "git+https://github.com/ultralytics/ultralytics.git@main"`
- Retry your workflow, ensuring the classification dataset path points to a directory (containing subfolders like /train and /val). Example:
  ```python
  from ultralytics import YOLO

  model = YOLO('yolo11n-cls.pt')
  model.train(data='path/to/classify_dataset')  # directory with /train and /val
  ```
- Alternatively, wait for the next official PyPI release and then:
  - `pip install -U ultralytics`

Feedback requested:
- Please let us know if the merged changes resolve the issue in your environment.
- If the problem persists, share a minimal snippet and the exact dataset path you’re passing (redact sensitive info).

Thanks for reporting this 🙏 and feel free to open follow-ups with any additional details.

**Comment by sashakeel** [comment]
Is it possible to use `data.yml` file for classification training?
My classes are defined as `[1, 2, 3, 4, 5, 6, 7, 8, 9, 10 ,11, 12,...N]` but the model due to lexicographic sorting treat them as `[1,11,12,...,2]` which is what I don't want.

**Comment by glenn-jocher** [comment]
Thanks for the question—classification doesn’t use a `data.yaml`; classes come from your subfolder names and are sorted lexicographically, so to keep numeric order either zero‑pad or prefix your folder names (e.g., `01, 02, …`) or override the mapping with a small custom trainer like this:

```python
from ultralytics import YOLO
from ultralytics.models.yolo.classify import ClassificationTrainer

class NumericOrderTrainer(ClassificationTrainer):
    def set_model_attributes(self):
        train_dir = self.data["train"]
        cls = [d.name for d in train_dir.iterdir() if d.is_dir()]
        names = dict(enumerate(sorted(cls, key=lambda s: int(s))))
        self.data["names"] = names
        self.model.names = names

model = YOLO("yolo11n-cls.pt")
model.train(data="path/to/dataset", trainer=NumericOrderTrainer, epochs=10, imgsz=224)
```

The lexicographic sort happens in the dataset check; see the check_cls_dataset reference for details: [classification dataset checker](https://docs.ultralytics.com/reference/data/utils/#ultralytics.data.utils.check_cls_dataset), and the trainer hook you’re overriding is documented here: [ClassificationTrainer reference](https://docs.ultralytics.com/reference/models/yolo/classify/train/).
